.changeset/chilled-teachers-bathe.md

```diff
@@ -0,0 +1,5 @@
+---
+'@as-integrations/aws-lambda': major
+---
+
+Implement strictly type request handlers, custom handlers, and expose middleware functionality
```

Comment:
This will be the note for the published CHANGELOG entry as well as the GH release. This should be a pretty at-length description of your changes. I think for the edge case / new stuff it's fine to point to readme (i.e. custom handler) but an example of what broke / how to transition v1 code to v2 should definitely go here.

You can probably take what you write here and make it your PR description + commit message as well, or they can at least be very similar.